### PR TITLE
fix: mypy - constrain `ignore_missing_imports = "True"`

### PR DESCRIPTION
**Pull request by motorina0**
### Summary
 - `pyproject.toml` had the flag `ignore_missing_imports = "True"`. It was too relaxed so all missing imports were ignored.
 - this PR narrows down what is ignored, enforces stricter rules and cleans-up redundant `# type: ignore`
 
### Changes

- [x] Explicitly specify the modules for which missing imports should be ignored:
  - there is a new config in `pyproject.toml` that specifies for which modules to ignore the imports
  - the global `mypy` `ignore_missing_imports = "True"` has been removed

- [x] use relative imports inside the extension module
  - this is enforced by `mypy` after the "explicit ignore" change
  - this is required for https://github.com/lnbits/lnbits/pull/1231
     - no absolute import of own modules:
        - **NOK**: `from lnbits.extensions.satspay.crud import check_address_balance, get_charge`
        - **OK**:    `from .crud import check_address_balance, get_charge, update_charge` 

- [x] remove redundant `# type: ignore`
  - there are now fewer ignores in the code
  - many of the remaining ones can be a source of bugs and should be handled
     - passing an `Optional` to a mandatory input param
     - importing from one extension from another (hard dependency, should use API calls) 
  - the bulk of the changed files are caused by this change.

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -96,6 +95,10 @@ exclude = """(?x)(
     | ^lnbits/wallets/lnd_grpc_files.
 )"""
 
+[[tool.mypy.overrides]]
+module = "embit.*,secp256k1.*,uvicorn.*,sqlalchemy.*,sqlalchemy_aio.*,websocket.*,websockets.*,pyqrcode.*,cashu.*,shortuuid.*,grpc.*,lnurl.*,bitstring.*,ecdsa.*,psycopg2.*,pyngrok.*"
```
Comment (motorina0) [inline pyproject.toml:99]: @dni is there a nicer way to build this list (eg: each module on a new line)?

Comment (dni) [reply pyproject.toml:99]: https://mypy.readthedocs.io/en/stable/config_file.html?highlight=pyproject#example-pyproject-toml

docs say its possible

```
[[tool.mypy.overrides]]
module = [
    "somelibrary",
    "some_other_library"
]
```

Comment (motorina0) [reply pyproject.toml]: updated, thanks!

**Review by dni — APPROVED** [review]
lgtm

**Pull request merged**